org.compiere.db.oracle.provider/src/org/compiere/db/DB_Oracle.java

```diff
@@ -950,6 +950,12 @@ public String getNameOfUniqueConstraintError(Exception e) {
 		return info.substring(fromIndex + 1, toIndex);
 	}
 
+	@Override
+	public String getForeignKeyConstraint(Exception e) {

```

Comment:
Please apply the following patch:
[pr2069PR.patch.txt](https://github.com/idempiere/idempiere/files/13114374/pr2069PR.patch.txt)

It implements oracle, and fixes error when postgresql is in German (for foreign and for unique discovery).

I also implemented a generic more explanatory message stating the table and column involved in the foreign key error.

Also, moved the code from PO to DBException, and added methods on CLogger to report the friendly message to user still saving the exception (there are plugins that uses the exception for other purposes).

Reply:
thanks Carlos, applied the patch, passed my basic tests